core/src/main/java/jenkins/util/AntClassLoader.java

```diff
@@ -1012,24 +1021,13 @@ protected URL getResourceURL(File file, String resourceName) {
                     }
                 }
             } else {
-                if (jarFile == null) {
-                    if (file.exists()) {
-                        jarFile = new JarFile(file);
-                        jarFiles.put(file, jarFile);
-                    } else {
-                        return null;
-                    }
-                    // potential race-condition
-                    jarFile = (JarFile) jarFiles.get(file);
-                }
-                JarEntry entry = jarFile.getJarEntry(resourceName);
-                if (entry != null) {
+                return storeAndConvert<URL>(jarFile, file, resourceName, (jar, entry) -> {
```

Comment:
compilation error. Should be ok with

```suggestion
                return storeAndConvert(jarFile, file, resourceName, (jar, entry) -> {
```

Reply:
This appears to be undoing the suggestion made by @oleg-nenashev which I applied as 6485ce053cb6d967fc9347846cb9af3aa1f2ab5e